### PR TITLE
Enabling automatic migration

### PR DESCRIPTION
**Pull request by milosevic81**
This is related to change merged in Azure/hydra-specs-pr#970

Purpose of this change is to add support for reading Service Tier Advisor
and Upgrade Hints for database, as well enable expanding request on
database and level, so we can enable automatic migration scenario.

Added RP wrapper for ServiceTierAdvisor
Refactoring of Recommended elastic pool
Added expand operation for Recommended elastic pool
Added expand operation on database level (expand hints and STA)

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@milosevic81__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by azurecla** [comment]
__@milosevic81__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

File: src/SiteRecoveryManagement/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@
 [assembly: AssemblyDescription("")]
 
 [assembly: AssemblyVersion("0.9.0.0")]
-[assembly: AssemblyFileVersion("0.5.1.0")]
+[assembly: AssemblyFileVersion("0.6.0.0")]
```
Comment (akromm) [inline src/SiteRecoveryManagement/Properties/AssemblyInfo.cs]: Do you really need to change all these assembly versions? Authentication, Compute, and SiteRecovery?

**Comment by yugangw-msft** [comment]
i will merge it or close it tomorrow, because this conflict with the ongoing work of moving hydra-specs-pr test projects

**Comment by ogail** [comment]
@azuresdkci test this please

**Comment by yugangw-msft** [comment]
Mine will cover this

**Pull request closed without merging**